src/encoded/viewconfigs/matrix.py

```diff
@@ -202,50 +196,60 @@ def preprocess_view(self):
                 type_info = self._types[self._doc_types[0]]
                 self._schema = type_info.schema
         self._validate_items(type_info)
-        self._result['title'] = self._set_result_title(type_info)
-        self._result['matrix'] = type_info.factory.matrix.copy()
-        self._matrix = self._result['matrix']
-        self._matrix['x']['limit'] = self._request.params.get('x.limit', 20)
-        self._matrix['y']['limit'] = self._request.params.get('y.limit', 5)
+        matrix = type_info.factory.matrix.copy()
+        matrix['x']['limit'] = self._request.params.get('x.limit', 20)
+        matrix['y']['limit'] = self._request.params.get('y.limit', 5)
         search_route = self._request.route_path('search', slash='/')
-        self._matrix['search_base'] = search_route + self._search_base
-        matrix_route = self._request.route_path('matrix', slash='/')
-        self._matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
-        self._result['views'] = self._construct_result_views(type_info)
-        query, used_filters = self._construct_query()
-        es_results = self._elastic_search.search(body=query, index=self._es_index)
+        matrix['search_base'] = search_route + self._search_base
+        matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
+        search_result = {
+            '@context': self._request.route_path('jsonld_context'),
+            'filters': [],
+            '@id': matrix_route + self._search_base,
+            '@type': ['Matrix'],
+            'notification': '',
+            'title': self._set_result_title(type_info),
+            'matrix': matrix,
+            'view': self._construct_result_views(type_info),
+        }
+        search_query, used_filters = self._construct_query(search_result, matrix)
+        es_results = self._elastic_search.search(
```

Comment:
You mentioned this in the ticket:

Second Note: Move see how much of the results creation could be moved below the es query search line. https://github.com/ENCODE-DCC/encoded/blob/dev/src/encoded/viewconfigs/matrix.py#L216

There is a Texas standoff situation where es_results needs search_result and search_result needs es_results; breaking this standoff could be tricky and not necessarily pretty. I recommend we leave it.

Reply:
Because of 'construct_query'?  I suggested looking into to why construct_query needs search_results and how that dependency was impacted by ENCD-4345/SNO-69.  I believe we've already fixed this.

But good job.  This is exactly what I wanted you to find per the Second Note.  A important step in cleaning up the back end matrix code is identifying unnecessary dependencies so we can fix them.  I think this is the line of code that lead us to correcting the set_filters via ENCD4345/SNO-69.

Reply:
I might can move the es_results above search_result. However, I think it would be better if I do that in a different ticket. The changes are tangential to ENCD-4353 (this ticket). 

How important is that to you? Two different tickets perform refactoring on matrix.py and adding a third ticket to the mix is doable if it is important.

Reply:
The only reason we're handing in 'search_result' to '_construct_query' is because 'set_filters' may add a 'filters' key with a list of terms dicts.  Let's make this dependency more clear by not passing in the final search_results dict.  Just pass in a dict with a 'filters' key.  

This will allow us to move the construction of the final search results down lower in the function.